File: csharp/Svix/Models/ClickhouseConfigIn.cs
// this file is @generated
using System.Text;
using Newtonsoft.Json;

namespace Svix.Models
{
    public class ClickhouseConfigIn
    {
        [JsonProperty("url", Required = Required.Always)]
        public required string Url { get; set; }

        [JsonProperty("username")]
        public string? Username { get; set; } = null;

        [JsonProperty("password")]
        public string? Password { get; set; } = null;

        [JsonProperty("database")]
        public string? Database { get; set; } = null;

        [JsonProperty("tableName", Required = Required.Always)]
        public required string TableName { get; set; }

        public override string ToString()
        {
            StringBuilder sb = new StringBuilder();

            sb.Append("class ClickhouseConfigIn {\n");
            sb.Append("  Url: ").Append(Url).Append('\n');
            sb.Append("  Username: ").Append(Username).Append('\n');
            sb.Append("  Password: ").Append(Password).Append('\n');
            sb.Append("  Database: ").Append(Database).Append('\n');
            sb.Append("  TableName: ").Append(TableName).Append('\n');
            sb.Append("}\n");
            return sb.ToString();
        }
    }
}

File: python/svix/models/clickhouse_config_in.py
# this file is @generated
import typing as t

from .common import BaseModel


class ClickhouseConfigIn(BaseModel):
    url: str
    """The HTTP URL of the ClickHouse server (e.g. `https://my_clickhouse:8443`)."""

    username: t.Optional[str] = None
    """Username to access Clickhouse.

    Currently a required field, but marked as optional because we may add different authentication in the future."""

    password: t.Optional[str] = None
    """Password to access Clickhouse.

    Currently a required field, but marked as optional because we may add different authentication in the future."""

    database: t.Optional[str] = None
    """The Clickhouse database to connect to."""

    table_name: str
    """The Clickhouse table to write to."""
